class/practice5.1/include/MegaData.hpp

```diff
@@ -0,0 +1,65 @@
+#include <print>
+#include <vector>
+#include <optional>
+class MegaData {
+public:
+    static const int smallArraySize = 1024;
+    static const int bigArraySize = 1024 * 1024;
+
+    float smallArray[smallArraySize];
+    double bigArray[bigArraySize];
+
+    MegaData() {
+        fillData();
+    }
+
+    void fillData() {
+        std::fill(smallArray, smallArray + smallArraySize, 42.0f);
+        std::fill(bigArray, bigArray + bigArraySize, 42.0);
+    }
+
+    bool operator==(const MegaData& other) const {
+        return this == &other;
+    }
+};
+
+class MegaDataPool {
+public:
+    MegaDataPool(int size) : pool(size), availableObjects(size, true) {}
+
+    int size() const {
+        return pool.size();
+    }
+
+    int usedSize() const {
+        return usedCounter;
```

Comment:
i hope we can remove usedCounter as we don't use it anywhere except this place. While for this function it is easy to calculate `pool.size() - available.size()` those calculations are fast as size function will be inlined as access to the corresponding field of vector.